test/unittest/readertest.cpp

```diff
@@ -190,7 +191,7 @@ static void TestParseDouble() {
         internal::Double e(x), a(h.actual_); \
         EXPECT_EQ(e.Sign(), a.Sign()); \
         if (fullPrecision) { \
-            EXPECT_EQ(x, h.actual_); \
+            EXPECT_NEAR(x, h.actual_, 0.0); \

```

Comment:
Here, we are in `fullPrecision`, so we should indeed check for full equality. Also you did disable the warning already.